frontstage/controllers/party_controller.py

```diff
@@ -304,8 +315,7 @@ def get_survey_list_details_for_party(party_id, tag, business_party_id, survey_i
     :survey_id: This is the surveys uuid
 
     """
-    enrolment_data = get_respondent_enrolments(party_id)
-
+    enrolment_data = list(get_respondent_enrolments(party_id))
```

Comment:
`get_respondent_enrolments` returns either a dictionary or None, is that correct?

Reply:
`get_respondent_enrolments` yields a generator (still don't get why we bothered with this) which is iterable already, so you shouldn't have to make it a list I think.
Is the adding of `list()` there to handle the occiasion when the respondent has no enrolments?

Reply:
It was made a list as it was getting a called twice. Under the hood it calls out to party . So making it a list means that we dont have so many http requests which are costly.

Reply:
Totally fair.  Could you do something for me and raise a card to look into changing that bit to not return a generator, but a normal list.  The return value for this function is fairly small so I don't understand why we were using yield to begin with.